File: client-scalaz/src/main/scala/org/openapitools/client/api/WordzWordResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WordzWordResponse._

case class WordzWordResponse (
  id: Option[Long],
word: Option[String],
definition: Option[String],
scores: Option[ScoreListResponse],
icon: Option[AssetShortResponse],
image: Option[AssetShortResponse],
authorOverride: Option[String],
updatedDate: Option[Long],
startDate: Option[Long],
endDate: Option[Long],
createdDate: Option[Long],
active: Option[Boolean],
allocateTickets: Option[Boolean],
ticketType: Option[String],
ticketCount: Option[Long],
points: Option[Long])

object WordzWordResponse {
  import DateTimeCodecs._

  implicit val WordzWordResponseCodecJson: CodecJson[WordzWordResponse] = CodecJson.derive[WordzWordResponse]
  implicit val WordzWordResponseDecoder: EntityDecoder[WordzWordResponse] = jsonOf[WordzWordResponse]
  implicit val WordzWordResponseEncoder: EntityEncoder[WordzWordResponse] = jsonEncoderOf[WordzWordResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/wordz_word_response.clj
(ns sirqul-io-t-platform.specs.wordz-word-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.score-list-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def wordz-word-response-data
  {
   (ds/opt :id) int?
   (ds/opt :word) string?
   (ds/opt :definition) string?
   (ds/opt :scores) score-list-response-spec
   (ds/opt :icon) asset-short-response-spec
   (ds/opt :image) asset-short-response-spec
   (ds/opt :authorOverride) string?
   (ds/opt :updatedDate) int?
   (ds/opt :startDate) int?
   (ds/opt :endDate) int?
   (ds/opt :createdDate) int?
   (ds/opt :active) boolean?
   (ds/opt :allocateTickets) boolean?
   (ds/opt :ticketType) string?
   (ds/opt :ticketCount) int?
   (ds/opt :points) int?
   })

(def wordz-word-response-spec
  (ds/spec
    {:name ::wordz-word-response
     :spec wordz-word-response-data}))
